src/Integration.UnitTests/Helpers/SonarLintOutputTests.cs

```diff
@@ -64,35 +63,95 @@ public void LogVerbose_OutputsToWindowIfLogLevelIsVerbose(DaemonLogLevel logLeve
-            serviceProviderMock.RegisterService(typeof(SVsOutputWindow), windowMock);
-
-            var sonarLintSettings = new Mock<ISonarLintSettings>();
-            sonarLintSettings.Setup(x => x.DaemonLogLevel).Returns(logLevel);
-
-            var logger = new SonarLintOutputLogger(serviceProviderMock, sonarLintSettings.Object);
+            var sonarLintSettings = CreateSonarLintSettings(logLevel);
 
-            logger.WriteLine("create window pane");
+            var testSubject = CreateTestSubject(serviceProviderMock, sonarLintSettings);
 
+            testSubject.WriteLine("create window pane");
             var outputPane = windowMock.AssertPaneExists(VsShellUtils.SonarLintOutputPaneGuid);
             outputPane.Reset();
 
             // Act
-            logger.LogVerbose("123 {0} {1}", "param 1", 2);
-            logger.LogVerbose("{0} {1} abc", 1, "param 2");
+            testSubject.LogVerbose("123 {0} {1}", "param 1", 2);
+            testSubject.LogVerbose("{0} {1} abc", 1, "param 2");
 
             // Assert
 
             if (shouldLogMessage)
             {
-                outputPane.AssertOutputStrings("DEBUG: 123 param 1 2", "DEBUG: 1 param 2 abc");
+                outputPane.AssertOutputStrings("[DEBUG] 123 param 1 2", "[DEBUG] 1 param 2 abc");
                 outputPane.AssertOutputStrings(2);
             }
             else
             {
                 outputPane.AssertOutputStrings(0);
             }
         }
+
+        [TestMethod]
+        [DataRow(DaemonLogLevel.Info)]
+        [DataRow(DaemonLogLevel.Minimal)]
+        [DataRow(DaemonLogLevel.Verbose)]
+        public void LogVerbose_PrefixIsAddedIfLogLevelIsVerbose(DaemonLogLevel logLevel)
+        {
+            // Arrange
+            var windowMock = new ConfigurableVsOutputWindow();
+            var serviceProviderMock = CreateConfiguredServiceProvider(windowMock);
+
+            var sonarLintSettings = CreateSonarLintSettings(logLevel);
+
+            var testSubject = CreateTestSubject(serviceProviderMock, sonarLintSettings);
+
+            testSubject.WriteLine("create window pane");
+            var outputPane = windowMock.AssertPaneExists(VsShellUtils.SonarLintOutputPaneGuid);
+            outputPane.Reset();
+
+            // Act
+            testSubject.WriteLine("writeline, no params");
+            testSubject.WriteLine("writeline, with params: {0}", "zzz");
+            testSubject.LogVerbose("verbose");
+
+            if (logLevel == DaemonLogLevel.Verbose)
+            {
+                outputPane.AssertOutputStrings(3);
+
+                var currentThreadId = System.Threading.Thread.CurrentThread.ManagedThreadId;
+                outputPane.AssertOutputStrings(
+                    $"[ThreadId {currentThreadId}] writeline, no params",
+                    $"[ThreadId {currentThreadId}] writeline, with params: zzz",
+                    $"[ThreadId {currentThreadId}] [DEBUG] verbose");
```

Comment:
With this change the thread id is written before all message when in `Verbose` mode e.g.

![image](https://user-images.githubusercontent.com/10757559/210617054-e4a738c2-0c91-44e6-a62f-dd811cc87408.png)